core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java

```diff
@@ -83,7 +83,7 @@ public String getName()
     {
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
-        Optional<TableHandle> tableHandle = plannerContext.getMetadata().getTableHandle(session, tableName);
+        Optional<TableHandle> tableHandle = plannerContext.getMetadata().getRedirectionAwareTableHandle(session, tableName).getTableHandle();
```

Comment:
@findepi / @losipiuk  : There are other tasks (e.g. : `io.trino.execution.DropTableTask`).
WDYT should I create a separate PR (and appropriate tests) to enhance the redirect functionality?

Reply:
https://github.com/trinodb/trino/pull/10577